src/plugins/intel_gpu/src/plugin/program.cpp

```diff
@@ -56,43 +57,74 @@ void Program::ValidateInputs(const std::shared_ptr<ngraph::Node>& op, std::vecto
-            std::dynamic_pointer_cast<ngraph::op::v3::NonMaxSuppression>(op) ||
-            std::dynamic_pointer_cast<ngraph::op::v1::NonMaxSuppression>(op) ||
-            std::dynamic_pointer_cast<ngraph::op::v0::PSROIPooling>(op) ||
-            std::dynamic_pointer_cast<ngraph::op::v0::ROIPooling>(op) ||
-            std::dynamic_pointer_cast<ngraph::op::v0::PriorBox>(op) ||
-            std::dynamic_pointer_cast<ngraph::op::v0::DetectionOutput>(op) ||
-            std::dynamic_pointer_cast<ngraph::op::v1::Reshape>(op) ||
-            std::dynamic_pointer_cast<ngraph::op::v0::Squeeze>(op) ||
-            std::dynamic_pointer_cast<ngraph::op::v0::Unsqueeze>(op) ||
-            std::dynamic_pointer_cast<ngraph::op::v1::Transpose>(op) ||
-            std::dynamic_pointer_cast<ngraph::op::v4::Proposal>(op) ||
-            std::dynamic_pointer_cast<ngraph::op::v0::Proposal>(op)) {
+auto getParamName = [](const std::shared_ptr<ov::Node>& param) -> std::string {
+    const auto& names = param->get_output_tensor(0).get_names();
+    if (!names.empty())
+        return *names.begin();
+    else
+        return param->get_friendly_name();
+};
+
+//  detect the only supported dynamic shape case -
+//  exactly one dimension is dynamic in input params with defined min/max interval
+bool Program::IsDynBatchModel(const std::shared_ptr<ov::Model>& model,
+                              std::map<std::string, ov::PartialShape>& shapes,
+                              std::map<std::string, std::pair<int64_t, int64_t>>& batch_dim) {
+    for (const auto& param : model->get_parameters()) {
+        auto pname = getParamName(param);
+        batch_dim[pname] = { -1, -1 };
+        if (param->get_output_partial_shape(0).rank().is_dynamic()) {
             return false;
         }
-
-        auto customLayer = m_config.customLayers.find(op->get_type_name());
-        if (customLayer != m_config.customLayers.end()) {
-            return false;
+        ov::PartialShape pshape = param->get_output_partial_shape(0);
+        int dynCount = 0;
+        int64_t batch_idx = -1;
+        for (size_t i = 0; i < pshape.size(); i++) {
+            if (pshape[i].is_dynamic()) {
+                dynCount++;
+                if (batch_idx < 0) {
+                    batch_idx = i;
+                }
+            }
+        }
+        switch (dynCount) {
+        case 1:
+            // exactly one dynamic dim
+            {
+                int64_t max_b = pshape[batch_idx].get_max_length();
+                if (max_b > 1) {
+                    batch_dim[pname].first = batch_idx;
+                    batch_dim[pname].second = pshape[batch_idx].get_max_length();
+                    pshape[batch_idx] = 1;
+                }
+            }
+        case 0:
+            // no dynamic dims - possible legacy case
+            shapes[pname] = pshape;
+            break;
+        default:
+            break;
```

Comment:
Is it correct that kind of exception isn't processed for "default" branch cases?

Reply:
It's incorrect to raise exception if this function is called from QueryNetwork.
We should just report that such dynamic layer is not supported.

In for only 1 dynamic dimension we report supported case to calling function.
For no dynamic dimension, we just fill parameter shapes for further analysis, to provide support for legacy dynamic batch.
For 2 or more dynamic dimensions, we report unsupported case.